### PR TITLE
Expire remaining 3.6 deprecations

### PR DESCRIPTION
**Pull request by ksunden**
## PR Summary

- Remaining mpl3.6 deprecations
- Update tests for deprecated behavior

I believe the only `3.6` deprecations that remain are still marked as "pending", those include some in axis3d and some in Figure.

The expiration for `Colorbar` filled kwarg is rather minimal.
It was originally deprecated in #21933, @jklymak do you think more clean up is needed?

## PR Checklist

**Documentation and Tests**
- [ ] Has pytest style unit tests (and `pytest` passes)
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] New plotting related features are documented with examples.

**Release Notes**
- [ ] New features are marked with a `.. versionadded::` directive in the docstring and documented in `doc/users/next_whats_new/`
- [ ] API changes are marked with a `.. versionchanged::` directive in the docstring and documented in `doc/api/next_api_changes/`
- [ ] Release notes conform with instructions in  `next_whats_new/README.rst` or `next_api_changes/README.rst`

### CODE REVIEW

File: lib/matplotlib/tri/_tripcolor.py [hunk 1]
```diff
@@ -80,10 +80,7 @@ def tripcolor(ax, *args, alpha=1.0, norm=None, cmap=None, vmin=None,
                 "tripcolor() missing 1 required positional argument: 'c'; or "
                 "1 required keyword-only argument: 'facecolors'")
         elif len(args) > 1:
-            _api.warn_deprecated(
-                "3.6", message=f"Additional positional parameters "
-                f"{args[1:]!r} are ignored; support for them is deprecated "
-                f"since %(since)s and will be removed %(removal)s")
+            raise TypeError(f"Unexpected positional parameters: {args[1:]!r}")
```
Comment (QuLogic) [inline lib/matplotlib/tri/_tripcolor.py:83]: Should use `_api.nargs_error`.

Comment (ksunden) [reply lib/matplotlib/tri/_tripcolor.py:83]: This is a confusing case for the "takes X, given Y" number of parameters. The expected number of parameters varies based on which call signature you are using, and those get normalized out by `Trangulation.get_from_args_and_kwargs` so `args` the variable needs to have only one value, but what you pass could be 

- 1: if you pass triangulation and facecolors(as kwarg)
- 2: either triangulation and c or x,y and facecolors(as kwarg)
- 3: x,y,c

In the cases where facecolors is passed, adding `c` will warn that it is ignored, but not error (and in fact, I'm realizing now will not error even if _more_ parameters are passed...) So the facecolors cases won't get to this codepath, which narrows it down to the triangulaion/c or x/y/c case, but we still don't know which version was used (without adding a boolean flag or something)

There is even a technical way to pass 4 parameters validly discussed (but discouraged) in the docstring.

Comment (QuLogic) [reply lib/matplotlib/tri/_tripcolor.py:83]: Well, `nargs_error` doesn't need to take a fixed number, but could take, e.g., `1-4` as the number of expected (which we do elsewhere already). However, it seems like you are saying that we're more specific here about extra arguments depending on other inputs, and don't want to limit to just the expected/actual message.

File: doc/api/next_api_changes/removals/25584-KS.rst [hunk 2]
```diff
@@ -0,0 +1,24 @@
+``Figure.callbacks`` is removed
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The Figure ``callbacks`` property is deprecated. The only signal was
```
Comment (QuLogic) [inline doc/api/next_api_changes/removals/25584-KS.rst:4]:
```suggestion
The Figure ``callbacks`` property has been removed. The only signal was
```

**Review by QuLogic — APPROVED** [review]

**Review by rcomer — APPROVED** [review]

**Comment by QuLogic** [comment]
They look like #25658.

File: lib/matplotlib/widgets.py [hunk 3]
```diff
@@ -2429,11 +2429,12 @@ def get_visible(self):
 
     @property
     def visible(self):
+        _api.warn_deprecated("3.8", alternative="get_visible")
         return self.get_visible()
 
     @visible.setter
     def visible(self, visible):
-        _api.warn_deprecated("3.6", alternative="set_visible")
+        _api.warn_deprecated("3.8", alternative="get_visible")
```
Comment (QuLogic) [inline lib/matplotlib/widgets.py:2437]:
```suggestion
        _api.warn_deprecated("3.8", alternative="set_visible")
```

File: doc/api/next_api_changes/removals/25584-KS.rst [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+The *filled* argument to ``Colorbar`` is removed
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+This behavior was already governed by the underlying ``ScalarMappable``.
+
+
+Widgets
+~~~~~~~
+
+The *visible* attribute setter of Selector widgets has been removed; use ``set_visible``
+The associated getter is also deprecated, but not yet expired.
```
Comment (QuLogic) [inline doc/api/next_api_changes/removals/25584-KS.rst:24]: Was this reverted now?

Comment (ksunden) [reply doc/api/next_api_changes/removals/25584-KS.rst:24]: ack, I neglected to to actually check what I said we were doing with this and was caught by a conflict, and resolved it differentely... re-settling on what I said earlier, as that was the leaning from conversation with Tom and Tim, I just forgot what was decided.

**Pull request merged**